src/electionguard_gui/components/key_ceremony_details_component.py

```diff
@@ -81,40 +94,19 @@ def on_key_ceremony_changed(self, key_ceremony_id: str) -> None:
         state = self._ceremony_state_service.get_key_ceremony_state(key_ceremony)
         self.log.debug(f"{key_ceremony_id} state = '{state}'")
         if is_admin and state == KeyCeremonyStates.PendingAdminAnnounce:
-            self.log.info("all guardians have joined, announcing guardians")
-            other_keys = self.announce(key_ceremony)
-            self.log.debug("saving other_keys")
-            self._key_ceremony_service.append_other_key(db, key_ceremony_id, other_keys)
-            # this notify_changed occurs inside watch_key_ceremonies and thus may
-            #       produce an unnecessary UI refresh for the admin
-            self._key_ceremony_service.notify_changed(db, key_ceremony_id)
-            # todo #689 wait until all guardians have backups
-
-        current_user_backups = len(key_ceremony.get_backups_for_user(current_user_id))
+            self._key_ceremony_s2_announce_service.run(db, key_ceremony)
```

Comment:
There was so much complex code in the key ceremony detail service, but now I'm finally starting to like the way this is broken out.  Still more refactoring to do, but it should be at least somewhat readable and maintainable now.